neotron-bmc-pico/src/main.rs

```diff
@@ -776,6 +845,59 @@ where
 				proto::Response::new_without_data(proto::ResponseResult::BadLength)
 			}
 		}
+		(proto::RequestType::Read, Ok(Command::SpeakerDuration)) => {
+			defmt::debug!("Reading speaker duration");
+			proto::Response::new_ok_with_data(unsafe {
+				&any_as_u8_slice(&register_state.speaker_config)[0..1]
```

Comment:
I'm not keen on unsafely casting objects as byte slices. I don't think the `Config` object is `repr(C)` so the fields aren't even guaranteed to be in the right order!

```suggestion
                                &[register_state.speaker.get_duration()]
				&any_as_u8_slice(&register_state.speaker_config)[0..1]
```

Reply:
Yeah, that was a bit of a hack to be able to provide a slice which doesn't point to a local var. But I'll just use a mutable var at the function scope.